libs/ledger-live-common/src/families/bitcoin/docs/BitcoinSync.md

```diff
@@ -0,0 +1,31 @@
+# Bitcoin Family Synchronization Logic in Ledger Live
+
+## Entry Point
+The synchronization logic for the Bitcoin family is primarily located in the `makeGetAccountShape` function within `js-synchronisation.ts`.
+
+## Main Data Structures
+Like other families in Ledger Live, each account is represented as an `Account`, as defined in `ledgerjs/packages/types-live/src/account.ts`. For the Bitcoin family, there's an additional `BitcoinResources` member within this `Account` to store Bitcoin-specific data structures. `BitcoinResources` includes a `walletAccount`, which contains information about the account's derivation path. Since a Bitcoin family account in Ledger Live is derived from `m / purpose' / coin_type' / account'` (BIP44), each account has a distinct derivation path. The `Xpub` data structure (`wallet-btc/xpub.ts` file) stores an account's information and synchronization status, mainly saved in the `storage` variable, including transactions and UTXOs for each address. Detailed data structure definitions are in `wallet-btc/storage/index.ts` file. To improve the efficiency of maintaining the transactions array, indexing has been added. To avoid recalculating the account's addresses, they are also cached.
+
+## Account Synchronization Logic
+The logic for synchronizing an account starts in the `sync()` function in `wallet-btc/xpub.ts`. Here are the steps:
+1. **Synchronize All Receiving Addresses:** Derived from `m / purpose' / coin_type' / account' / 0`.
+2. **Synchronize All Change Addresses:** Derived from `m / purpose' / coin_type' / account' / 1`.
+3. **Batch Synchronization:** In both step 1 and 2, addresses are synchronized in batches of 20. For example, receiving addresses from `m / purpose' / coin_type' / account' / 0 / 0` to `m / purpose' / coin_type' / account' / 0 / 19` are synchronized first. If any address in these 20 has a transaction, the next batch of 20 addresses is synchronized, continuing until a batch with no transactions is found.
+4. **Synchronization Details:**
+    - Addresses are obtained using `this.crypto.getAddress(this.derivationMode, this.xpub, account, index)`.
+    - All confirmed transactions for an address are fetched from `https://explorers.api.live.ledger.com/blockchain/v4/[coin_ticker]/address/[address]/txs`.
+    - All pending transactions are fetched from `https://explorers.api.live.ledger.com/blockchain/v4/[coin_ticker]/address/[address]/txs/pending`.
+    - All transactions for an address are summarized and UTXOs are calculated and stored in `storage`.
```

Comment:
Maybe just a bit of information on how we estimate balance based on utxos data ?